src/components/CustomTable/StyledTableBody/StyledTableBody.js

```diff
@@ -17,12 +17,23 @@ const useStyles = makeStyles((theme) => ({
   },
 }))
 
-const StyledTableBody = ({ columns, data, onRowClick }) => {
+const StyledTableBody = ({
+  columns,
+  data,
+  onRowClick,
+  tablePage,
+  rowsPerPage,
+}) => {
   const classes = useStyles()
 
+  const slicedRows = data.slice(
+    tablePage * rowsPerPage,
+    tablePage * rowsPerPage + rowsPerPage
+  )
```

Comment:
You should be doing this form your parent component, the table should only receive an array of data to show